### PR TITLE
[New Entry] JavaScript Strings: `.valueOf()`

### PR DESCRIPTION
**Pull request by dorian-edwards**
### Description

Created Term Entry for JavaScript Strings method .valueOf() per issue #590 by adding new file valueOf.md in new folder, valueOf, within terms

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by Dusch4593** [comment]
Hello @dorian-edwards and thank you for this contribution to Docs! 🚀 I'm gonna take a look at this for the first review and leave you some comments/suggestions after I'm finished. I'll tag you again once I'm done. 😄

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:15]:
```suggestion
The **`.valueOf()`** method returns the primitive value of a `String` object, similar to the `toString()` method of JavaScript Strings. This method is not used explicitly in code, but rather behind the scenes within JavaScript.

> **Note:** This method is for `String` objects and not to be confused with the `.valueOf()` method of the `Object` class.
```

For Docs entries, we prefer to **bold** the first mention of a topic, concept, or term at hand. Also, is there a way to unpack this sentence a little more?

> This method is not used explicitly in code, but rather behind the scenes within JavaScript.

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:20]:
```suggestion
stringObject.valueOf();
```

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:22]:
```suggestion

The `.valueOf` method takes no parameters.

```

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:25]:
```suggestion
All primitive string values are wrapped in a `String` object (`new String(value)`). In the following example, this primitive value is returned by the `.valueOf()` method:
```

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:32]:
```suggestion

This will print the following:

```

And then add a \```shell snippet that looks like this:

```shell
[String: 'Bill']
Bill
```

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:30]:
```suggestion
console.log(name);
console.log(name.valueOf());
```

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:41]:
```suggestion
const animal = 'cat';
console.log(animal === animal.valueOf());
console.log('apple'.valueOf() === 'apple');
```

No need for output comments in the Codebyte snippets since they're runnable and the output will be rendered.

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:35]:
```suggestion
A less common use case for the `.valueOf()` method is calling on variables with string literals or on the string literal directly:
```

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:37]:
```suggestion
```codebyte/javascript
```

The signatures for Codebyte snippets are a little different. More details [here](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md#codebytes-runnable-code-blocks).

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:9]:
```suggestion
  - 'Functions'
  - 'Methods'
  - 'Objects'
  - 'Strings'
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@dorian-edwards Finished with my review and left you some comments/suggestions. Let me know if you have any questions or when you're finished pushing the new changes. 😄

**Comment by yangc95** [comment]
Hi @dorian-edwards, I'll be taking this PR on for a second review!

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:17]:
```suggestion
The **`.valueOf()`** method returns the [primitive](https://www.codecademy.com/resources/docs/javascript/data-types) value of a `String` object, similar to the `.toString()` method of JavaScript Strings. Usually this method isn't used in code but is called internally by JavaScript.
```

Comment (yangc95) [reply content/javascript/concepts/strings/terms/valueOf/valueOf.md:17]: Primitive here is referring to a primitive data type, or a value that is not an object (no methods/properties). Thinking that by linking the data types entry, this will be more clear.

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:19]:
```suggestion
> **Note:** This `String` method is not to be confused with the `.valueOf()` method that exists for the `Object` class.
```

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:27]:
```suggestion
The `.valueOf()` method takes no parameters.
```

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:48]:
```suggestion
A less common use case for the `.valueOf()` method is to call it on variables with string literals or on the string literal directly:
```

Comment (yangc95) [reply content/javascript/concepts/strings/terms/valueOf/valueOf.md:48]: Can we also include a sentence about what the common use case for this method is?

**Review by yangc95 — COMMENTED** [review]
@dorian-edwards Conducted the second review for this. Please tag me once you're complete or if you have any additional questions prior to editing.

Thank you!

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (dorian-edwards) [reply content/javascript/concepts/strings/terms/valueOf/valueOf.md]: @yangc95 I did as much digging as I could but all I could manage to find were vague references to how JavaScript uses this method 'internally' or 'automatically' and was not typically used, nor needed, explicitly in code. This includes text references in addition to digital ones. Do you know of where I can find additional information?

**Comment by dorian-edwards** [comment]
Hey @yangc95, just checking if there's something more I need to do before this can be merge?

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:19]: Can we use a different perspective for this note and say something like:

"Objects that are descendants of the `Object` class, including `String`s, inherit the `valueOf()` method."

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:17]: Thanks for looking into common use case of this method! Since we have this last sentence here, could we rephrase the preface to the code byte below? (see below)

```suggestion
The **`.valueOf()`** method returns the [primitive](https://www.codecademy.com/resources/docs/javascript/data-types) value of a `String` object, similar to the `.toString()` method of JavaScript Strings. Usually this method is called internally by JavaScript.
```

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:48]:
```suggestion
Although the `.valueOf()` method is typically invoked internally by JavaScript, it can be used by calling it on a `String` object:
```

**Comment by yangc95** [comment]
> Hey @yangc95, just checking if there's something more I need to do before this can be merge?

@dorian-edwards Thanks for making those edits! I had a few more comments, after which I feel this may be merge-ready. LMK if you have any further questions!

**Comment by dorian-edwards** [comment]
@yangc95 All done! Thanks for the feedback and giving me the chance to contribute!

File: content/javascript/concepts/strings/terms/valueOf/valueOf.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/valueOf/valueOf.md:17]:
```suggestion
The **`.valueOf()`** method returns the [primitive](https://www.codecademy.com/resources/docs/javascript/data-types) value of a `String` object, similar to the `.toString()` method. Usually, this method is called internally by JavaScript.
```

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
TY @dorian-edwards I made a quick change in wording here [c6a6398](https://github.com/Codecademy/docs/pull/1023/commits/c6a63985b2e6ef34e2428ad9159f3a874567c1dc)

Here is the live link!
https://www.codecademy.com/resources/docs/javascript/strings/valueOf

**Pull request merged**
